modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/prepare/PlannerHelper.java

```diff
@@ -97,8 +98,10 @@ public static IgniteRel optimize(SqlNode sqlNode, IgnitePlanner planner) {
 
             joinSizeFinder.visit(rel);
 
-            if (joinSizeFinder.sizeOfBiggestJoin() > MAX_SIZE_OF_JOIN_TO_OPTIMIZE) {
-                Set<String> disabledRules = new HashSet<>(HintUtils.disabledRules(root.hints));
+            boolean amountOfJoinsAreBig = joinSizeFinder.sizeOfBiggestJoin() > MAX_SIZE_OF_JOIN_TO_OPTIMIZE;
```

Comment:
Seems more clear to move it into separate function isn`t it ?

Reply:
agree